Toms-Storage-120/src/shared/java/com/tom/storagemod/gui/AbstractStorageTerminalScreen.java

```diff
@@ -257,7 +255,11 @@ protected void updateSearch() {
 				for (int i = 0;i < getMenu().itemListClient.size();i++) {
 					StoredItemStack is = getMenu().itemListClient.get(i);
 					if (is != null && is.getStack() != null) {
-						String dspName = searchMod ? BuiltInRegistries.ITEM.getKey(is.getStack().getItem()).getNamespace() : is.getStack().getHoverName().getString();
+						String dspName;
+						if (searchMod) dspName = BuiltInRegistries.ITEM.getKey(is.getStack().getItem()).getNamespace();
+						else if (searchNbt && is.getStack().hasTag()) dspName = String.valueOf(is.getStack().getTag());
```

Comment:
Use `NbtUtils.prettyPrint` instead of toString on NBTs
Implement the same caching system as with tootips (it's nbt sensitive)
https://github.com/tom5454/Toms-Storage/blob/1.20.x/Toms-Storage-120/src/shared/java/com/tom/storagemod/gui/AbstractStorageTerminalScreen.java#L60-L67

Reply:
Also added Tag caching at the same occasion

Reply:
The cache should be used here